line_obfuscator.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	mathrand "math/rand"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+const (
+	// PosMax is the largest line or column value that can be represented without loss.
+	// Source: https://golang.org/src/cmd/compile/internal/syntax/pos.go
+	PosMax = 1 << 30
+
+	// PosMin is the smallest correct value for the line number.
+	// Source: https://github.com/golang/go/blob/2001685ec01c240eda84762a3bc612ddd3ca93fe/src/cmd/compile/internal/syntax/parser_test.go#L229
+	PosMin = 1
+)
+
+func prependComment(group *ast.CommentGroup, comment *ast.Comment) *ast.CommentGroup {
+	if group == nil {
+		return &ast.CommentGroup{List: []*ast.Comment{comment}}
+	}
+
+	group.List = append([]*ast.Comment{comment}, group.List...)
+	return group
+}
+
+func clearCommentGroup(group *ast.CommentGroup) *ast.CommentGroup {
```

Comment:
Either leave a comment or rename this function to make it more clear that compiler directives are not cleared